BraveRewardsUI/Wallet/WalletViewController.swift

```diff
@@ -364,3 +343,81 @@ class WalletViewController: UIViewController, RewardsSummaryProtocol {
     }
   }
 }
+
+extension WalletViewController {
+  
+  func startNotificationObserver() {
+    // Stopping as a precaution
+    stopNotificationObserver()
+    // Add observer
+    NotificationCenter.default.addObserver(self, selector: #selector(notificationAdded(_:)), name: NSNotification.Name.BATBraveLedgerNotificationAdded, object: nil)
+    loadNextNotification()
+  }
+  
+  func stopNotificationObserver() {
+    NotificationCenter.default.removeObserver(self, name: NSNotification.Name.BATBraveLedgerNotificationAdded, object: nil)
+    currentNotification = nil
+  }
+  
+  @objc private func notificationAdded(_ notification: Notification) {
+    // TODO: Filter notification?
+    // LoadNext if there is no current notification
+    if currentNotification == nil {
+      loadNextNotification()
+    }
+  }
+  
+  private func clearCurrentNotification() {
+    if let currentNotification = currentNotification {
+      state.ledger.clearNotification(currentNotification)
+      self.currentNotification = nil
+    }
+  }
+  
+  private func loadNextNotification() {
+    //.failedContribution, .adsLaunch, .invalid & .backupWallet are not supported right now
+    // TODO: Verify that notifications are ordered & they are displayed in FIFO.
+    let ignoredRewardNotificationTypes: [RewardsNotification.Kind] = [.failedContribution,
+                                                                      .invalid,
+                                                                      .backupWallet,
+                                                                      .adsLaunch]
+    if let notification = state.ledger.notifications.first(where: {
+      !ignoredRewardNotificationTypes.contains($0.kind)
+    }) {
+      currentNotification = notification
+      if let notificationView = RewardsNotificationViewBuilder.get(notification: notification) {
+        notificationView.closeButton.addTarget(self, action: #selector(tappedNotificationClose), for: .touchUpInside)
+        (notificationView as? WalletActionNotificationView)?.actionButton.addTarget(self, action: #selector(tappedNotificationAction), for: .touchUpInside)
+        walletView.setNotificationView(notificationView, animated: true)
+      } else {
+        clearCurrentNotification()
+        loadNextNotification()
+      }
+    } else {
+      walletView.setNotificationView(nil, animated: true)
+    }
+  }
+  
+  @objc private func tappedNotificationClose() {
+    clearCurrentNotification()
+    loadNextNotification()
+  }
+  
+  @objc private func tappedNotificationAction() {
+    if let notification = currentNotification {
+      switch notification.kind {
+      case .grant, .grantAds:
+        tappedGrantsButton()
+      case .pendingNotEnoughFunds, .insufficientFunds:
+        tappedAddFunds()
```

Comment:
We actually _cant_ have the button here for add funds 😞

Since wallet management isn't in scope for this release despite the view already made and working, we'll have to make these message/alert notifications I guess?